src/ucp/core/ucp_context.c

```diff
@@ -883,14 +884,17 @@ ucp_is_resource_in_transports_list(const char *tl_name,
         tmp_tl_cfg_mask = 0;
         if (ucp_tls_array_is_present(alias->tls, alias_arr_count, tl_name, info,
                                      &tmp_rsc_flags, &dummy_mask)) {
+            aux_tl = tmp_rsc_flags & UCP_TL_RSC_FLAG_AUX;
             if (ucp_config_is_tl_name_present(allow_list, alias->alias, 1,
                                               &tmp_rsc_flags,
                                               &tmp_tl_cfg_mask)) {
                 if (allow_list->mode == UCS_CONFIG_ALLOW_LIST_ALLOW) {
                     *rsc_flags   |= tmp_rsc_flags;
                     *tl_cfg_mask |= tmp_tl_cfg_mask;
                     return 1;
-                } else {
+                } else if (!aux_tl) {
```

Comment:
why not just checking `tmp_rsc_flags & UCP_TL_RSC_FLAG_AUX` here?

Reply:
from the other side `AUX` flags can also be added in `ucp_config_is_tl_name_present``on line 895 and we will miss it

Reply:
Yes, but this is not what we want to check. We need only results from `ucp_tls_array_is_present`, because it checks presence of `tl_name` in transport layers belong to the current alias. And `ucp_config_is_tl_name_present` checks presence of alias name in allow/deny list.